Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -410,6 +438,25 @@ NAKED void thread_context_first_enter(void)
         "b restore_context_and_eret \n");
 }
 
+NAKED void do_assume_context(Thread*, u32)
+{
+    // clang-format off
+    asm(
+        "mov x19, x0 \n" // save thread ptr
+        // We're going to call Processor::init_context, so just make sure
+        // we have enough stack space so we don't stomp over it
+        "sub sp, sp, #" __STRINGIFY(8 + REGISTER_STATE_SIZE + TRAP_FRAME_SIZE + 8) " \n"
+        "bl do_init_context \n"
+        "mov sp, x0 \n"  // move stack pointer to what Processor::init_context set up for us
+        "mov x0, x19 \n" // to_thread
+        "mov x1, x19 \n" // from_thread
+        "sub sp, sp, 24 \n"
```

Comment:
can this magic number could be related to something more specific (as above)?

Reply:
Not really, what it's related to is the fact that I do 2 stores after it, but I can't really relate the line to lines that come after it :slightly_smiling_face:

Reply:
🙇🏻‍♂️